src/wrappers/request.js

```diff
@@ -1,40 +1,53 @@
+
 import http from 'http';
 import https from 'https';
-import url from 'url';
-import httpProxyAgent from 'http-proxy-agent';
-import httpsProxyAgent from 'https-proxy-agent';
-import { request } from 'axios';
-import { getProxyForUrl } from 'proxy-from-env';
+import urlUtil from 'url';
 
-export default function(options, cb) {
-  const requestOptions = {
-    url: options.uri,
-    headers: options.headers,
-    timeout: options.timeout
-  };
+export default (options) => {
+  if (options.fetcher) {
+    return options.fetcher(options.uri);
+  }
 
-  const proxyUrl = options.proxy || getProxyForUrl(options.uri);
-  if (proxyUrl || options.agentOptions || options.strictSSL != undefined) {
-    const agentOptions = {
-      ...(options.strictSSL != undefined) && { rejectUnauthorized: options.strictSSL },
-      ...(options.headers && { headers: options.headers }),
-      ...options.agentOptions
-    };
+  return new Promise((resolve, reject) => {
+    const {
+      host,
+      path,
+      port,
+      protocol
+    } = urlUtil.parse(options.uri);
 
-    if (proxyUrl) {
-      // Axios proxy workaround: https://github.com/axios/axios/issues/2072
-      const proxyOptions = url.parse(proxyUrl);
-      requestOptions.proxy = false; //proxyParsed
-      const proxyAgentOptions = { ...agentOptions, ...proxyOptions };
-      requestOptions.httpAgent = new httpProxyAgent(proxyAgentOptions);
-      requestOptions.httpsAgent = new httpsProxyAgent(proxyAgentOptions);
-    } else {
-      requestOptions.httpAgent = new http.Agent(agentOptions);
-      requestOptions.httpsAgent = new https.Agent(agentOptions);
-    }
-  }
+    const requestOptions = {
+      host,
+      path,
+      port,
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
```

Comment:
the request doesn't have any specific content type, as it's a get and bears no body. You might've thought to have an Accept header but even then the application/json is speculative as per the JWK spec its meant to be `application/jwk-set+json`. Best to leave both content-type and accept headers out completely.